packages/commerce-sdk-react/src/hooks/useDNT.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {useState} from 'react'
+import useAuthContext from './useAuthContext'
+import {getDefaultCookieAttributes} from '../utils'
+import Cookies from 'js-cookie'
+
+interface useDntReturn {
+    dntNotSet: boolean
+    updateDNT: (preference: boolean | null) => Promise<void>
+}
+
+/**
+ * Hook that updates the DNT preference and refreshes access token.
+ *
+ * @group Helpers
+ * @category Shopper Authentication
+ *
+ */
+const useDNT = (): useDntReturn => {
+    const auth = useAuthContext()
+
+    const dwDntValue = Cookies.get('dw_dnt')
+    const dntCookieIsDefined = dwDntValue !== '1' && dwDntValue !== '0'
+    const [dntNotSet, setDntNotSet] = useState(dntCookieIsDefined)
```

Comment:
```suggestion
    const [dntStatus, setDntStatus] = useState(dntCookieIsDefined)
```

Reply:
This is probably related to the misnaming of `dntCookieIsDefined` which I'll fix to `dntCookieIsNotDefined`

This boolean is supposed to indicate the dnt preference not being set yet, so the consent tracking UI can be rendered based on it. It might be best to keep some element of 'Not set' in the name of the boolean